app/server/src/main/scala/org/bitcoins/server/BitcoindRpcAppConfig.scala

```diff
@@ -138,40 +151,55 @@ case class BitcoindRpcAppConfig(
   lazy val zmqConfig: ZmqConfig =
     ZmqConfig(zmqHashBlock, zmqRawBlock, zmqHashTx, zmqRawTx)
 
-  lazy val bitcoindInstance: BitcoindInstance = {
-    val fallbackBinary =
-      if (isRemote) BitcoindInstance.remoteFilePath else DEFAULT_BINARY_PATH
-
-    BitcoindInstance(
-      network = network,
-      uri = uri,
-      rpcUri = rpcUri,
-      authCredentials = authCredentials,
-      zmqConfig = zmqConfig,
-      binary = binaryOpt.getOrElse(fallbackBinary),
-      datadir = bitcoindDataDir,
-      isRemote = isRemote,
-      proxyParams = socks5ProxyParams
-    )
+  lazy val bitcoindInstance = binaryOpt match {
+    case Some(file) =>
+      BitcoindInstanceLocal(
+        network = network,
+        uri = uri,
+        rpcUri = rpcUri,
+        authCredentials = authCredentials,
+        zmqConfig = zmqConfig,
+        binary = file,
+        datadir = bitcoindDataDir
+      )
+
+    case None =>
+      BitcoindInstanceRemote(network = network,
+                             uri = uri,
+                             rpcUri = rpcUri,
+                             authCredentials = authCredentials,
+                             zmqConfig = zmqConfig,
+                             proxyParams = socks5ProxyParams)
   }
 
   lazy val client: BitcoindRpcClient = {
-    val version = versionOpt.getOrElse(bitcoindInstance.getVersion)
-    implicit val system: ActorSystem =
-      ActorSystem.create("bitcoind-rpc-client-created-by-bitcoin-s", config)
-    BitcoindRpcClient.fromVersion(version, bitcoindInstance)
-  }
+    val version: Option[BitcoindVersion] = bitcoindInstance match {
+      case local: BitcoindInstanceLocal =>
+        Some(versionOpt.getOrElse(local.getVersion))
+      case _: BitcoindInstanceRemote => None
+
+    }
+
+    bitcoindInstance match {
+      case _: BitcoindInstanceLocal =>
+        BitcoindRpcClient.fromVersion(version.get, bitcoindInstance)
+      case _: BitcoindInstanceRemote => new BitcoindRpcClient(bitcoindInstance)
+    }
```

Comment:
shouldn't need to do a match here twice can just do

```scala
bitcoindInstance match {
      case local: BitcoindInstanceLocal =>
        val version = versionOpt.getOrElse(local.getVersion)
        BitcoindRpcClient.fromVersion(version.get, bitcoindInstance)
      case _: BitcoindInstanceRemote => 
        new BitcoindRpcClient(bitcoindInstance)
    }

```

Reply:
Oh yeah, I'll change that